src/screens/loading.rs

```diff
@@ -6,11 +6,14 @@ use bevy::prelude::*;
 use crate::{asset_tracking::ResourceHandles, screens::Screen, theme::prelude::*};
 
 pub(super) fn plugin(app: &mut App) {
-    app.add_systems(OnEnter(Screen::Loading), spawn_loading_screen);
+    app.add_systems(
+        OnEnter(Screen::Loading),
+        spawn_loading_screen.run_if(not(all_assets_loaded)),
```

Comment:
```suggestion
        spawn_loading_screen.run_if(not(all_assets_loaded)),
```
I'd rather move this check to `splash::continue_to_title_screen` (renamed ofc). It's better to skip the loading screen entirely if there's nothing to load, instead of flashing the app's clear color for 1 frame.

Reply:
Hmm? The splash does not really have anything to do with the loading screen now. This check here is for when the player presses `Esc` to restart the game, which bypasses the splash entirely

Reply:
My bad, replace `splash::continue_to_title_screen` with `title::enter_loading_screen`. 

(also off-topic, I noticed we're naming the systems `continue_to` vs. `enter` when it may be reasonable to name them all `enter`).

Reply:
Done: https://github.com/TheBevyFlock/bevy_new_2d/pull/373/commits/b11db8a317ce1d1a9919ecb4c8c2a16e4b597e3b